### DISCUSSION TITLE
how to import types in v6?

### DISCUSSION THREAD

**Discussion by wuxingping** [fabricjs/fabric.js#8715] [General]
i imported types in v6 as follow,but there was an error `Module '"fabric"' has no exported member 'TClassProperties'.ts(2305)`:

import { FabricObject} from 'fabric/dist/src/shapes/Object/FabricObject';
import { TClassProperties } from 'fabric/src/typedefs';

so i donot know how to import types in v6?

**Comment by robothot** [comment]
see https://github.com/fabricjs/fabric.js/pull/8717

**Reply by wuxingping** [reply]
i saw it,thank you so much.but i have been using v6.maybe should commit to merge on branch 6.0.0-beta1 as well

**Reply by robothot** [reply]
see https://github.com/fabricjs/fabric.js/blob/master/package.json#L5 

master branch is 6.0.0-beta1

**Reply by wuxingping** [reply]
> see https://github.com/fabricjs/fabric.js/blob/master/package.json#L5
> 
> master branch is 6.0.0-beta1

okay,that's great.

**Reply by wuxingping** [reply]
> see https://github.com/fabricjs/fabric.js/blob/master/package.json#L5
> 
> master branch is 6.0.0-beta1

it is still v5(5.3.0) when i run` npm install fabric`.the v6, isn't it master?i thought if i performed `npm install fabric`,it would download the version 6.0.0-beta1.so now i have to run ` npm install fabric@6.0.0-beta1` as ever

**Comment by iansama** [comment]
where is the solution to the problem? this conversation doesnt help OP's problem

**Reply by zhe-he** [reply]
```ts
import { Rect } from "fabric"

import type { Rect } from "fabric";
```

**Reply by asturur** [reply]
Correct.
Also it may happen some types are not exported, by mistake, and so we need to export them after we get to know the issue.
